test/js/node/stream/node-stream.test.js

```diff
@@ -588,3 +588,11 @@ it("should send Readable events in the right order", async () => {
     ``,
   ]);
 });
+
+it("emits newListener event _before_ adding the listener", () => {
+  const stream = new Stream();
```

Comment:
This test would not fail if `newListener` is never called

We need to assert that the event handler has been called

```suggestion
  const cb = jest.fn((event) => {
        expect(stream.listenerCount(event)).toBe(0);
  });
  const stream = new Stream();
  stream.on("newListener", cb);
  stream.on("foo", () => {});
  expect(cb).toHaveBeenCalled();
```